### PR TITLE
Add tests for systemd units

### PR DESCRIPTION
**Pull request by tescande**

### CODE REVIEW

File: tests/fs_diff/test_fs_diff.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import subprocess
+
+# Requirements:
+# - 2 XCP-ng host of same version
+
+pytest.fixture(scope='module')
+def test_fs_diff(hosts):
+    assert len(hosts) == 2, "This test requires exactly 2 hosts"
+
+    assert (hosts[0].xcp_version == hosts[1].xcp_version), f"Hosts versions differ"
```
Comment (stormi) [inline tests/fs_diff/test_fs_diff.py:12]: The message associated to an assert should state what is expected, not what's wrong when it fails.

File: jobs.py [hunk 2]
```diff
@@ -407,6 +407,15 @@
         "params": {},
         "paths": ["tests/pci_passthrough"],
     },
+    "fs-diff": {
+        "description": "Check for difference between 2 hosts file system",
+        "requirements": [
+            "2 XCP-ng host >= 8.2"
+        ],
+        "nb_pools": 1,
```
Comment (stormi) [inline jobs.py:415]: Two here.

File: tests/systemd/test_systemd.py
Comment (stormi) [inline tests/systemd/test_systemd.py:1]: Is this test related to all the other commits in this PR?

File: tests/systemd/test_systemd.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        logging.error(f"Unit {unit.split()[0]} failed")
+
+    assert not failed_services
+
+pytest.fixture(scope='module')
+def test_unit_dep_cycles(host):
+    analyse = host.ssh(['systemd-analyze', 'verify', 'default.target'])
+    err = False
+    for line in analyse.splitlines():
+        m = re.match("^Found ordering cycle on .*$", line)
```
Comment (stormi) [inline tests/systemd/test_systemd.py:22]: Is this the only kind of error `systemd-analyze verify` could reveal?

File: jobs.py [hunk 4]
```diff
@@ -416,6 +416,15 @@
...
     },
+    "systemd": {
+        "description": "Testing systemd failed units and dependency cycles",
+        "requirements": [
+            "A XCP-ng host >= 8.2"
+        ],
+        "nb_pools": 1,
+        "params": {},
+        "paths": ["tests/systemd"],
+    },
```
Comment (stormi) [inline jobs.py:427]: I'd rather have it in the main job, not separated into its own job. Also, I'm not sure we need a new `systemd` directory in `tests`. `tests/system` could be a fit.

File: tests/systemd/test_systemd.py (hunk 3 above)
Comment (tescande) [reply tests/systemd/test_systemd.py:22]: Sure not, but in this test `test_unit_dep_cycles()` we're only look for dep cycle issues.

File: jobs.py (hunk 4 above)
Comment (tescande) [reply jobs.py:427]: Works for me. Will move it into system (it was there in the first place actually :))

File: tests/fs_diff/test_fs_diff.py (hunk 1 above)
Comment (tescande) [reply tests/fs_diff/test_fs_diff.py]: Fixed

File: jobs.py [hunk 5]
```diff
@@ -407,6 +407,15 @@
         "params": {},
         "paths": ["tests/pci_passthrough"],
     },
+    "fs-diff": {
+        "description": "Check for difference between 2 hosts file system",
```
[the remark this answers is not in the document]
Comment (tescande) [reply jobs.py]: Fixed

File: tests/systemd/test_systemd.py
Comment (tescande) [reply tests/systemd/test_systemd.py:1]: Moved to a new PR: https://github.com/xcp-ng/xcp-ng-tests/pull/246

File: jobs.py (hunk 4 above)
Comment (tescande) [reply jobs.py]: Fixed

File: tests/systemd/test_systemd.py (hunk 3 above)
Comment (tescande) [reply tests/systemd/test_systemd.py]: Will investigate other useful systemd tests

**Review by ydirson — DISMISSED** [review]
Rather than starting a blacklist with one failure case we happen to know about (cycles), what about instead using a whitelist with the specific warnings we did evaluate as being acceptable?
